evennia/contrib/collab/commands.py

```diff
@@ -0,0 +1,817 @@
+
+    @desc/long object
+    """
+    key = '@desc'
+    aliases = ['@describe']
+
+    def func(self):
+        if not ((self.lhs or self.rhs) or 'long' in self.switches):
+            self.msg("{rUsage: @desc thing=Description{n")
+        obj = self.caller.search(self.lhs)
+        if not obj:
+            return
+        if not collab_check(self.caller, obj, locks=['desc']):
+            self.msg("rYou don't have permission to describe this object.")
+            return
+        if self.rhs and 'long' in self.switches:
+            self.msg("{rYou may specify a description, or use the long flag, "
+                     "but not both.{n")
+            return
+        if self.rhs:
+            obj.db.desc = self.rhs
+            self.msg("{gDescription set.")
+            return
+        # Time to call in the line editor.
+
+        # hook save/load functions
+        def load():
+            return obj.db.desc or ""
+
+        def save():
+            """
+            Save line buffer to given attribute name. This should
+            return True if successful and also report its status.
+            """
+            obj.db.desc = self.editor.buffer
+            self.caller.msg("Saved.")
+            return True
+
+        self.editor = LineEditor(self.caller,
+                                 loadfunc=load,
+                                 savefunc=save,
+                                 key="desc")
+
+
+class CmdCollabCpAttr(CmdCpAttr):
+    def check_from_attr(self, obj, attr, clear=False):
+        attr_name, handler = prefix_check(obj, attr)
+        if clear:
+            access = 'write'
+        else:
+            access = 'read'
+        if not attr_check(self.caller, obj, access, handler):
+            if clear:
+                action = 'clear'
+            else:
+                action = 'read from'
+            self.msg("{rYou are not allowed to %s '%s' on '%s'.{n"
+                     % (action, attr, obj))
+            return False
+        return True
+
+    def check_to_attr(self, obj, attr):
+        attr_name, handler = prefix_check(obj, attr)
+        if not attr_check(self.caller, obj, 'write', handler):
+            self.msg("{rYou are not allowed to write to '%s' on '%s'."
+                     % (obj, attr))
+            return False
+        return True
+
+    def check_has_attr(self, obj, attr):
+        attr_name, handler = prefix_check(obj, attr)
+        if not handler.has(attr_name):
+            self.caller.msg(
+                "%s doesn't have an attribute %s."
+                % (obj.name, attr))
+            return False
+        return True
+
+    def get_attr(self, obj, attr):
+        """
+        Hook for overriding on subclassed commands. Do any preprocessing
+        required and get the attribute from the object.
+        """
+        attr_name, handler = prefix_check(obj, attr)
+        return handler.get(attr_name)
+
+
+class CmdExamine(MuxCommand):
```

Comment:
Interesting. Re-implementing the examine command like this is a good exercise but may be hard to consistently maintain. I wonder if one could maybe make the main command more pluggable. The ability to selectively show various sections (as shown here) could also be more generalized in core I reckon.